earthly/python/Earthfile

```diff
@@ -26,8 +29,16 @@ POETRY_SETUP:
     #   "Connection pool is full, discarding connection: pypi.org. Connection pool size: 10"
     RUN poetry config installer.max-workers 10
 
+# Common python builder setup.
+# Parameters:
+#   opts      : Options passed to the `poetry install` command.
+BUILDER:
+    COMMAND
+    ARG opts
+
     # Copy our dependencies.
-    COPY pyproject.toml poetry.lock $extra_file .
+    COPY pyproject.toml .
+    COPY --if-exists poetry.lock .
```

Comment:
`poetry lock` MUST exist.  If it doesn't the python project has not properly vendored its dependencies.
Also, why did you remove the ability to copy an extra file here?

Reply:
To have a generalized builder I have followed these suggestions https://python-poetry.org/docs/basic-usage/#as-a-library-developer. 
But if you think that we should include this file every time and we are not going to have python libs, I will reverse this change.

Reply:
>why did you remove the ability to copy an extra file here?

I have removed it from this place because it will be much easier to COPY needed files outside the UDC command, 
as it was done here https://github.com/input-output-hk/catalyst-ci/blob/48502f89eed3af79af5ed3641f92b9f35c89674e/examples/python/Earthfile#L6 . UDC arguments could not provide so wide functionality in the convenient way (like passing some options to `COPY` command)

Reply:
Yes, we should enforce a vendored and uptodate lock file for now.  If it turns out later we don't need it we can worry about that then.  I don't plan on us making python libraries anytime soon.  We only use python for Automation, testing and "glue" tasks, all of which are executables.

Reply:
I am ok with the removal of extra file copy.

Reply:
updated